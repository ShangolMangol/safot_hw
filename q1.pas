program PascalTriangle;

type LineArray = array[1..15] of integer;

procedure CreateNextLine(var line: LineArray; var prevLine: LineArray; level: integer);
    var i: integer;
    begin
        i := 2;
        if level <> 1 then
        begin    
            repeat
                line[i] := prevLine[i-1] + prevLine[i];
                i := i + 1;
            until (i > level);
        end;
    end;

var n, i, j: integer;
var line, prevLine: LineArray;

begin
for i := 1 to 15 do
    begin
        line[i] := 1;
        prevLine[i] := 1
    end;
ReadLn(n);
for i := 1 to n do
    begin
        for j := 1 to i-1 do
        begin
            write(line[j], ' ');
        end;
        writeLn(line[i]);
        
        if i <> 15 then
        begin
            for j := 1 to 15 do
            begin
                prevLine[j] := line[j]
            end;
            
            CreateNextLine(line, prevLine, i)

        end;
    end;
end.
