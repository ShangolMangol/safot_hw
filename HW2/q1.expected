val it = () : unit
val it = fn : 'a -> 'b -> ('a * 'b -> 'b) -> 'b
val it = fn : int * real -> (real -> string) -> bool
val it = fn : ('a -> 'b -> 'c) -> 'a -> 'b -> 'd -> 'c
val it = fn : 'a -> 'b -> int -> int -> int
val it = fn : ('a -> 'b) -> 'a -> ('b * 'b -> 'c) -> 'c
val it = fn : unit -> unit -> int
val it = fn : 'a -> 'a * 'a -> 'a
val it = fn : int * string * string -> int * string * string
- 
